### PR TITLE
feat: Implement missing methods in product module and make more tests pass

### PR DESCRIPTION
**Pull request by sradevski**
The 2 bigger remaining tasks are:
1. handling prices for variants
2. Handling options (breaking change)

After that all tests should pass on both v1 and v2

### CODE REVIEW

File: packages/product/src/joiner-config.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export const joinerConfig: ModuleJoinerConfig = {
       },
     },
     {
-      name: ["variant", "variants"],
+      name: ["producyt_variant", "product_variants", "variant", "variants"],
```
Comment (olivermrbl) [inline packages/product/src/joiner-config.ts]:
```suggestion
      name: ["product_variant", "product_variants", "variant", "variants"],
```

File: packages/medusa/src/api-v2/admin/products/[id]/variants/route.ts [hunk 2]
```diff
@@ -56,5 +58,5 @@ export const POST = async (
     throw errors[0].error
   }
 
-  res.status(200).json({ product_variant: result[0] })
+  res.status(200).json({ variant: result[0] })
```
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/products/[id]/variants/route.ts:61]: question: As per our recent discussion, should we return product here?

**Review by olivermrbl — APPROVED** [review]
Solid work 💪

File: packages/medusa/src/api-v2/admin/products/[id]/variants/route.ts (hunk 2 above)
Comment (sradevski) [reply packages/medusa/src/api-v2/admin/products/[id]/variants/route.ts:61]: @olivermrbl yes there is a test for it that is skipped, once enabled I'll update this, I'll review all routes at that time to make sure we are consistent

File: packages/product/src/joiner-config.ts (hunk 1 above)
Comment (sradevski) [reply packages/product/src/joiner-config.ts]: gosh, nice catch 😄

**Pull request merged**